daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/SequenceParserBases.scala

```diff
@@ -227,12 +227,18 @@ abstract class SequenceParserBase(
               pstate.setSuccess()
               isDone = true
             }
+
+            // We successfully parsed a discriminator, but failed to parse the discriminated content.
+            // Do not continue trying to parse other memebers of the unordered sequence.
+            case DiscriminatedFailure if (!isOrdered) => isDone = true
```

Comment:
If I am correct above and DiscriminatedFailure is only for unordered case, then I think the if(!isUnordered) should not be a conditional branch here, but an Assert.invariant(isUnordered) before the assignment of isDone = true.

Reply:
Fixed.